plugins/inputs/mysql/v2/convert.go

```diff
@@ -53,6 +55,69 @@ func ParseString(value sql.RawBytes) (interface{}, error) {
 	return string(value), nil
 }
 
+// ParseWsrepLatency parses the given sql.RawBytes value into a map
+// containing 5 distinct float64 values. These represent min/avg/max/stdev/sample_size.
+// It returns an error if the value is unrecognized.
+func parseWsrepLatency(value sql.RawBytes) (interface{}, error) {
+	parts := strings.Split(string(value), "/")
+	if len(parts) != 5 {
+		return nil, fmt.Errorf("unsupported amount of values in wsrep_evs_repl_latency, got %d, should be 5", len(parts))
+	}
+
+	keys := []string{"min", "avg", "max", "stdev", "sample_size"}
+	result := make(map[string]interface{}, len(keys))
+
+	for i, key := range keys {
+		val, err := strconv.ParseFloat(parts[i], 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse %s value: %w", key, err)
+		}
+		result[key] = val
+	}
+
+	return result, nil
+}
+
+var wsrespGcacheSizeRe = regexp.MustCompile(`(\d+)([MG]?)`)
+
+// ParseWsrepProviderOptions parses the given sql.RawBytes value into a map
+// containing all wsrep Provider options settings.
+func parseWsrepProviderOptions(value sql.RawBytes) (interface{}, error) {
+	parts := strings.Split(string(value), "; ")
+
+	result := make(map[string]interface{})
+
+	for _, setting := range parts {
+		splitSetting := strings.Split(setting, " = ")
+
+		// only process gcache.size for now
+		if splitSetting[0] != "gcache.size" {
+			continue
+		}
+
+		key := strings.Replace(splitSetting[0], ".", "_", -1)
+
+		data := wsrespGcacheSizeRe.FindStringSubmatch(splitSetting[1])
+
+		var val float64
+		val, err := strconv.ParseFloat(data[1], 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse %s value: %w", data[1], err)
+		}
+
+		switch data[2] {
+		case "M":
+			val = val * 1024 * 1024
+		case "G":
+			val = val * 1024 * 1024 * 1024
+		}
+
+		result[key] = val
+	}
+
+	return result, nil
+}
```

Comment:
Can we please move this to a separate PR!?

Reply:
Done.